resharper/src/resharper-unity/Rider/UnitySolutionTracker.cs

```diff
@@ -5,40 +5,79 @@
 using JetBrains.Platform.RdFramework.Base;
 using JetBrains.Platform.RdFramework.Util;
 using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.Unity.ProjectModel;
+using JetBrains.Util;
 
 namespace JetBrains.ReSharper.Plugins.Unity.Rider
 {
     [SolutionComponent]
     public class UnitySolutionTracker
     {
         private readonly ISolution mySolution;
-        public readonly RProperty<bool> IsAbleToEstablishProtocolConnectionWithUnity = new RProperty<bool>();
+        public readonly RProperty<bool> IsUnityProjectFolder = new RProperty<bool>();
+        public readonly RProperty<bool> IsUnityGeneratedProject = new RProperty<bool>();
+        public readonly RProperty<bool> IsUnityProject = new RProperty<bool>();
 
-        public UnitySolutionTracker(ISolution solution, IFileSystemTracker fileSystemTracker, Lifetime lifetime, IShellLocks locks)
+        public UnitySolutionTracker(ISolution solution, IFileSystemTracker fileSystemTracker, Lifetime lifetime, IShellLocks locks,
+            UnityHost unityHost, UnityReferencesTracker unityReferencesTracker)
         {
             mySolution = solution;
-            
             if (locks.Dispatcher.IsAsyncBehaviorProhibited) // for tests
                 return;
 
-            IsAbleToEstablishProtocolConnectionWithUnity.SetValue(
-                ProjectExtensions.IsAbleToEstablishProtocolConnectionWithUnity(mySolution.SolutionDirectory));
-            
+            SetValues();
+
             fileSystemTracker.AdviseDirectoryChanges(lifetime, mySolution.SolutionDirectory.Combine(ProjectExtensions.AssetsFolder), false,
                 OnChangeAction);
-            fileSystemTracker.AdviseDirectoryChanges(lifetime, mySolution.SolutionDirectory.Combine(ProjectExtensions.LibraryFolder), false,
-                OnChangeAction);
-            fileSystemTracker.AdviseDirectoryChanges(lifetime, mySolution.SolutionDirectory.Combine(ProjectExtensions.ProjectSettingsFolder), false,
-                OnChangeAction);
+            fileSystemTracker.AdviseDirectoryChanges(lifetime, mySolution.SolutionDirectory.Combine(ProjectExtensions.ProjectSettingsFolder), true,
```

Comment:
This `true` means to watch recursively, right? Do we need it `true` for `ProjectSettings`?

Reply:
Ah, wait, is this so we see changes to `ProjectVersion.txt` or `*.asset`, too? If so, might be worth a comment as to why.

Reply:
True. Recursive for  `ProjectVersion.txt` or `*.asset`